hil/api.py

```diff
@@ -800,7 +800,10 @@ def network_create(network, owner, access, net_id):
     else:
         if not get_network_allocator().validate_network_id(net_id):
             raise BadArgumentError("Invalid net_id")
-        allocated = False
+        if not get_network_allocator().network_id_available(net_id):
+            raise AllocationError("Network ID is in use. Please choose a "
+                                  "different net_id")
+        allocated = get_network_allocator().is_network_id_in_pool(net_id)
```

Comment:
At this point in the code, this is still always going to be False. Two sensible options:

* change this back to `allocated = False`
* move it out of the else block, so that it is executed regardless of which branch is taken, and remove `allocated = True` from the first branch.